apps/ledger-live-mobile/src/newArch/features/ModularDrawer/hooks/useDetailAccount.ts

```diff
@@ -0,0 +1,125 @@
+  if (a) return -1;
+  if (b) return 1;
+  return 0;
+};
+
+export const useDetailedAccounts = (
+  asset: CryptoOrTokenCurrency,
+  flow: string,
+  source: string,
+  accounts$?: Observable<WalletAPIAccount[]>,
+) => {
+  const state = useCountervaluesState();
+  const { trackModularDrawerEvent } = useModularDrawerAnalytics();
+  const navigation = useNavigation();
+  const accountIds = useGetAccountIds(accounts$);
+  const nestedAccounts = useSelector(accountsSelector);
+  const counterValueCurrency = useSelector(counterValueCurrencySelector);
+
+  const isATokenCurrency = useMemo(() => isTokenCurrency(asset), [asset]);
+
+  const accounts = useMemo(() => {
+    const accountTuples = getAccountTuplesForCurrency(asset, nestedAccounts, accountIds);
+    return orderBy(accountTuples, [(tuple: AccountTuple) => tuple.account.balance], ["desc"]);
+  }, [asset, nestedAccounts, accountIds]);
+
+  const overridedAccountName = useBatchMaybeAccountName(accounts.map(({ account }) => account));
+
+  const detailedAccounts = useMemo(() => {
+    const accountNameMap = keyBy(
+      accounts
+        .map(({ account }, index) => ({
+          id: account.id,
+          name: overridedAccountName[index],
+        }))
+        .filter(item => item.name),
+      "id",
+    );
+
+    const formattedAccounts = accounts.map(tuple => {
+      const account = tuple.account;
+      const protocol = getTagDerivationMode(account.currency, account.derivationMode) ?? "";
+      const currencyAccount = formatDetailedAccount(
+        account,
+        account.freshAddress,
+        state,
+        counterValueCurrency,
+      );
+
+      if (isATokenCurrency && tuple.subAccount) {
+        const formattedSubAccount = formatDetailedAccount(
+          tuple.subAccount,
+          account.freshAddress,
+          state,
+          counterValueCurrency,
+        );
+        const parentAccountName = accountNameMap[account.id]?.name;
+        return {
+          ...formattedSubAccount,
+          name: parentAccountName ?? formattedSubAccount.name,
+        };
+      } else {
+        const accountName = accountNameMap[account.id]?.name;
+        return {
+          ...currencyAccount,
+          protocol,
+          name: accountName ?? currencyAccount.name,
+        };
+      }
+    });
+
+    return sortAccountsByFiatValue(formattedAccounts);
+  }, [accounts, state, counterValueCurrency, isATokenCurrency, overridedAccountName]);
+
+  const onAccountClick = useCallback(
+    (account: AccountUI) => {
+      trackModularDrawerEvent("button_clicked", {
+        button: "Add a new account",
+        page: MODULAR_DRAWER_PAGE_NAME.MODULAR_ACCOUNT_SELECTION,
+        flow,
+        source,
+      });
+      navigation.navigate(NavigatorName.Accounts, {
+        screen: ScreenName.Account,
+        params: {
+          currencyId: asset.id,
+          accountId: account.id,
+        },
+      });
+    },
+    [trackModularDrawerEvent, flow, source, navigation, asset.id],
+  );
```

Comment:
Hmmm imho account should be selected (from onAccountSelected callbacl), close the drawer but it doesn't lead to the Account detail page